index.js

```diff
@@ -102,6 +104,57 @@ function fastifyPostgres (fastify, options, next) {
     }
   }
 
+  fastify.addHook('onRoute', routeOptions => {
+    const useTransaction = routeOptions.useTransaction
+
+    const transactionFailedSymbol = Symbol('transactionFailed')
+
+    if (useTransaction) {
+      // This will rollback the transaction if the handler fails at some point
+      const onError = async (req, reply, error) => {
+        req[transactionFailedSymbol] = true
+
+        try {
+          await req.pg.query('ROLLBACK')
+        } catch (err) {
+          // await req.pg.query('ROLLBACK')
+        }
+      }
```

Comment:
please check in a unit test where you mock req.pg.query what happens if sending the ROLLBACK query fails, because according to the fastify docs, the onError hook does not support sending an error to the done callback

```suggestion
      const onError = (req, reply, error, done) => {
        req[transactionFailedSymbol] = true

        req.pg.query('ROLLBACK', done)
      }
```